src/common/perf.py

```diff
@@ -200,9 +200,36 @@ def report_nodes_perf(self):
         # Currently reporting one metric per node
         for node in self.all_reports:
             # CPU UTILIZATION
+            cpu_avg_utilization = [ report["cpu_pct_per_cpu_avg"] for report in self.all_reports[node] ]
+
             self.metrics_logger.log_metric(
                 "max_t_(cpu_pct_per_cpu_avg)",
-                max([ report["cpu_pct_per_cpu_avg"] for report in self.all_reports[node] ]),
+                max(cpu_avg_utilization),
+                step=node
+            )
+            self.metrics_logger.log_metric(
+                "cpu_avg_utilization_pct",
+                sum(cpu_avg_utilization)/len(cpu_avg_utilization),
+                step=node
+            )
+            self.metrics_logger.log_metric(
+                "cpu_avg_utilization_at100_pct",
+                sum( [ utilization >= 100.0 for utilization in cpu_avg_utilization])/len(cpu_avg_utilization)*100.0,
+                step=node
+            )
+            self.metrics_logger.log_metric(
+                "cpu_avg_utilization_over80_pct",
+                sum( [ utilization >= 80.0 for utilization in cpu_avg_utilization])/len(cpu_avg_utilization)*100.0,
+                step=node
```

Comment:
Is there any way to make it clean to users about "step=node"?

Reply:
There's no way currently. Ideally, I'd want to store a table with 2 columns, one the metric value, second the node. But this is currently not possible with mlflow, so I'm using this trick to record the node id. We'll have to rely on documentation to help clarify.